src/main/java/domain/piece/Cha.java

```diff
@@ -0,0 +1,63 @@
+package domain.piece;
+
+import static domain.board.Direction.DOWN;
+import static domain.board.Direction.LEFT;
+import static domain.board.Direction.RIGHT;
+import static domain.board.Direction.UP;
+
+import domain.board.Board;
+import domain.board.Direction;
+import domain.board.Node;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cha implements Piece {
+
+    private static final List<Direction> CHA_MOVABLE_DIRECTIONS = List.of(UP, RIGHT, DOWN, LEFT);
+
+    private final Team team;
+
+    public Cha(final Team team) {
+        this.team = team;
+    }
+
+    @Override
+    public List<Node> findMovableNodes(final Node sourceNode, final Board board) {
+        List<Node> candidates = new ArrayList<>();
+        CHA_MOVABLE_DIRECTIONS.stream()
+                .filter(sourceNode::hasEdgeByDirection)
+                .forEach(direction ->
+                        findCandidates(sourceNode.findNextNodeByDirection(direction), board, direction, candidates));
+        return candidates;
+    }
+
+    private void findCandidates(Node currentNode,
+                                final Board board, final Direction direction,
+                                final List<Node> candidates) {
+        if (!board.hasPieceTeamByNode(currentNode, this.team)) {
+            candidates.add(currentNode);
+        }
+
+        while (true) {
+            if (!currentNode.hasEdgeByDirection(direction)) {
+                break;
+            }
```

Comment:
```suggestion
        while (currentNode.hasEdgeByDirection(direction)) {
```
while 문의 조건을 더 명확하게 표현해보면 좋을 것 같네요

Reply:
재귀호출을 while로 바꾸면서 체크하지 못한 부분 중 하나네요🥲
while문의 조건을 명확히 해주는 것은 무한루프를 방지할 수 있어서 좋다고 생각합니다!